File: backend/Services/Parsers/SedecOrderParser.cs
using System.Text.RegularExpressions;
using backend.Models;

namespace backend.Services.Parsers
{
    public class SedecOrderParser : BasicScoreResultParser
    {
        private ILogger<SedecOrderParser> _logger;

        public SedecOrderParser(ILogger<SedecOrderParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override bool CountWinner => false;
        public override bool Default => false;
        public override string GameName => "Sedec-order";
        public override string? HelpText => null;
        protected override Regex Parser => new Regex($@"Daily {GameName} #\d+\nGuesses: (?<{ScoreGroup}>(\d\d?)|X).*?{ExtraContent}", RegexOptions.Singleline);
        protected override string? ExtraContent => "https://sedecordle.com\n#sedecordle  #sedecorder";
        public override string Url => "https://www.sedecordle.com/sedec-order";

        public override string? GetScoreValue(DailyResult dailyResult)
        {
            if (dailyResult.Scores == null)
            {
                return null;
            }

            return "\"" + string.Join(",", Enumerable.Range(0, 16).ToArray()
                .Select((i) => {
                    if (dailyResult.Scores.Count <= i)
                    {
                        return "X";
                    }

                    return dailyResult.Scores[i].ToString();
                })) + "\"";
        }
    }
}

File: lambdas/src/parsers/sedecorder-parser.ts
import { DailyResult } from '../models/daily-result';
import { BasicScoreResultParser } from './base/basic-score-parser';
import { getGroup, hasGroup } from './base/result-parser';

export class SedecOrderParser extends BasicScoreResultParser {
  readonly countWinner = true;
  readonly gameName = 'Sedec-order';
  readonly helpText = null;
  protected readonly extraContent = 'https://sedecordle.com/\n#sedecordle  #sedecorder';
  readonly url = 'https://www.sedecordle.com/sedec-order';
  
  // Matches "Daily Sedec-order #1087\nGuesses: 18/21" or "X/21"
  // C# regex: `Daily {GameName} #\d+\nGuesses: (?<score>(\d\d?)|X).*?{ExtraContent}` with Singleline flag
  protected readonly parser = new RegExp(
    `Daily ${this.gameName} #\\d+\\nGuesses: (?<${BasicScoreResultParser.SCORE_GROUP}>(\\d\\d?)|X).*?https://sedecordle.com/\\n#sedecordle  #sedecorder`,
    's'
  );
  
  getScoreValue(dailyResult: DailyResult): string | null {
    if (!dailyResult.scores) return null;
    
    const values = Array.from({ length: 16 }, (_, i) => {
      if (dailyResult.scores && dailyResult.scores.length > i) {
        return dailyResult.scores[i].toString();
      }
      return 'X';
    });
    
    return `"${values.join(',')}"`;
  }
}
